exetera/core/fields.py

```diff
@@ -143,6 +162,110 @@ def _ensure_valid(self):
         if not self._valid_reference:
             raise ValueError("This field no longer refers to a valid underlying field object")
 
+    def where(self, cond:Union[list, tuple, np.ndarray, Field, Callable], b, inplace=False):
+        if isinstance(cond, (list, tuple, np.ndarray)):
+            cond = cond
+        elif isinstance(cond, Field):
+            if cond.indexed:
+                raise NotImplementedError("Where does not support indexed string fields at present")
+            cond = cond.data[:]
+        elif callable(cond):
+            cond = cond(self.data[:])
+        else:
+            raise TypeError("'cond' parameter needs to be either callable lambda function, or array like, or NumericMemField")
+
+        result_mem_field = None
+
+        if isinstance(self, IndexedStringField) and isinstance(b, IndexedStringField):
```

Comment:
When doing the type checking need to check that it's one of two types: ` isinstance(self, (IndexedStringField, IndexedStringMemField))`.